<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>GeneratorDialog</class>
 <widget class="QDialog" name="GeneratorDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>340</width>
    <height>298</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Create a new map</string>
  </property>
  <property name="modal">
   <bool>true</bool>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="0" column="0">
    <widget class="QLabel" name="label">
     <property name="text">
      <string>Map width:</string>
     </property>
    </widget>
   </item>
   <item row="0" column="1">
    <widget class="QSpinBox" name="m_width">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="toolTip">
      <string>Map width in meters. Must be a multiple of 100 m.</string>
     </property>
     <property name="suffix">
      <string> m</string>
     </property>
     <property name="minimum">
      <number>1100</number>
     </property>
     <property name="maximum">
      <number>5000</number>
     </property>
     <property name="singleStep">
      <number>100</number>
     </property>
     <property name="value">
      <number>1100</number>
     </property>
    </widget>
   </item>
   <item row="1" column="0">
    <widget class="QLabel" name="label_2">
     <property name="text">
      <string>Map height:</string>
     </property>
    </widget>
   </item>
   <item row="1" column="1">
    <widget class="QSpinBox" name="m_height">
     <property name="toolTip">
      <string>Map height in meters. Must be a multiple of 100 m.</string>
     </property>
     <property name="suffix">
      <string>  m</string>
     </property>
     <property name="minimum">
      <number>800</number>
     </property>
     <property name="maximum">
      <number>3000</number>
     </property>
     <property name="singleStep">
      <number>100</number>
     </property>
     <property name="value">
      <number>800</number>
     </property>
    </widget>
   </item>
   <item row="2" column="1">
    <widget class="QCheckBox" name="m_autoGenerate">
     <property name="toolTip">
      <string>Populate the new map with randomized trees and woods</string>
     </property>
     <property name="text">
      <string>Auto generate</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item row="3" column="0">
    <widget class="QLabel" name="label_3">
     <property name="text">
      <string>Woods:</string>
     </property>
    </widget>
   </item>
   <item row="3" column="1">
    <widget class="QSpinBox" name="m_woods">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="toolTip">
      <string>The woods/trees coverage % for the new map</string>
     </property>
     <property name="suffix">
      <string> %</string>
     </property>
     <property name="minimum">
      <number>0</number>
     </property>
     <property name="maximum">
      <number>75</number>
     </property>
     <property name="value">
      <number>30</number>
     </property>
    </widget>
   </item>
   <item row="4" column="0">
    <widget class="QLabel" name="label_4">
     <property name="text">
      <string>Octave:</string>
     </property>
    </widget>
   </item>
   <item row="4" column="1">
    <widget class="QSpinBox" name="m_octave">
     <property name="toolTip">
      <string>Wooded area roughness</string>
     </property>
     <property name="minimum">
      <number>1</number>
     </property>
     <property name="maximum">
      <number>16</number>
     </property>
     <property name="value">
      <number>2</number>
     </property>
    </widget>
   </item>
   <item row="5" column="0">
    <widget class="QLabel" name="label_5">
     <property name="text">
      <string>Frequency:</string>
     </property>
    </widget>
   </item>
   <item row="5" column="1">
    <widget class="QSpinBox" name="m_frequency">
     <property name="toolTip">
      <string>Wooded area finer roughness</string>
     </property>
     <property name="minimum">
      <number>1</number>
     </property>
     <property name="maximum">
      <number>8</number>
     </property>
     <property name="value">
      <number>5</number>
     </property>
    </widget>
   </item>
   <item row="6" column="0">
    <widget class="QLabel" name="label_6">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="text">
      <string>Rivers:</string>
     </property>
    </widget>
   </item>
   <item row="6" column="1">
    <widget class="QSpinBox" name="m_rivers">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="maximum">
      <number>10</number>
     </property>
     <property name="value">
      <number>1</number>
     </property>
    </widget>
   </item>
   <item row="7" column="1">
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>36</height>
      </size>
     </property>
    </spacer>
   </item>
   <item row="8" column="0">
    <spacer name="horizontalSpacer">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>141</width>
       <height>20</height>
      </size>
     </property>
    </spacer>
   </item>
   <item row="8" column="1">
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>GeneratorDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_autoGenerate</sender>
   <signal>toggled(bool)</signal>
   <receiver>m_woods</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>255</x>
     <y>86</y>
    </hint>
    <hint type="destinationlabel">
     <x>251</x>
     <y>114</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_autoGenerate</sender>
   <signal>toggled(bool)</signal>
   <receiver>m_octave</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>255</x>
     <y>86</y>
    </hint>
    <hint type="destinationlabel">
     <x>251</x>
     <y>148</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_autoGenerate</sender>
   <signal>toggled(bool)</signal>
   <receiver>m_frequency</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>255</x>
     <y>86</y>
    </hint>
    <hint type="destinationlabel">
     <x>251</x>
     <y>182</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
